lib/sync-DataSetModel.js

```diff
@@ -375,43 +429,70 @@ var self = {
     datasetClient.syncPending = false;
     datasetClient.syncLoopStart = new Date().getTime();
 
+    async.waterfall(
+      [
+        function (next) {
+          self.doListHandler(dataset.id, datasetClient.queryParams, next, datasetClient.metaData);
+        },
+        function (records, next) {
+          self.generateRecordHashes(dataset.id, records, next);
+        },
+        function (hashes, next) {
+          var globalHash = hashes.hash;
+
+          var previousHash = datasetClient.dataHash ? datasetClient.dataHash : '<undefined>';
+          syncUtil.doLog(dataset.id, 'verbose', 'doSyncList cb ' + ( cb !== undefined) + ' - Global Hash (prev :: cur) = ' + previousHash + ' ::  ' + globalHash);
+
+          datasetClient.dataHash = globalHash;
+          self.datasetClientRecords[datasetClient.id] = hashes.records;
+
+          datasetClient.syncRunning = false;
+          datasetClient.syncLoopEnd = new Date().getTime();
+
+          next(null, hashes);
+        }
+      ],
+      function (err, hashes) {
+        if (err) {
+          datasetClient.syncRunning = false;
+          datasetClient.syncLoopEnd = new Date().getTime();
+          return cb && cb(err);
+        } else {
+          if (cb) {
+            cb(null, hashes);
+          }
+        }
+      }
+    )
+
     self.doListHandler(dataset.id, datasetClient.queryParams, function (err, records) {
```

Comment:
@evanshortiss I think we should remove this function? Looks like it's a duplication of what you have added above?